### PR TITLE
feat: toNumber built-in function

### LINKED ISSUE #584: add built-in $toNumber(..)
**Is your feature request related to a problem? Please describe.**
If we retrieve an element of the API response that is a number, e.g. `additions`, this will be a string. We don't have a way to then compare it.

**Describe the solution you'd like**
Allow the possibility to do `$toNumber($selectFromContext("$.additions")) > 10` in Aladino specifications.

### PR DESCRIPTION
**Pull request by zolamk**
## Description
Adds a new built-in function `$toNumber`.

### Examples
- `$toNumber("5")`
- `$toNumber($selectFromContext("$.commits"))`

## Related issue

Closes #584 

## Type of change

New feature (non-breaking change which adds functionality)

## How was this tested?
Unit tests

## Checklist

- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works 
- [x] I have ran `task check -f` and have no issues

## Code review and merge strategy (ship/show/ask) 

 
 
- [x] Ask: this pull request requires a code review before merge

### CODE REVIEW

File: plugins/aladino/functions/toNumber_test.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+func TestToNumber(t *testing.T) {
+	tests := map[string]struct {
+		str     string
+		wantRes aladino.Value
+		wantErr error
+	}{
+		"when empty": {
+			str: "",
+			wantErr: fmt.Errorf(`error converting "" to number: %w`, &strconv.NumError{
+				Func: "Atoi",
```
Comment (ferreiratiago) [inline plugins/aladino/functions/toNumber_test.go:29]: Why is this `Func` needed?

File: plugins/aladino/functions/toNumber_test.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+				Num:  "",
+				Err:  errors.New("invalid syntax"),
```
Comment (ferreiratiago) [inline plugins/aladino/functions/toNumber_test.go:31]: It there any reason for this to be capitalized? Can we put it in lowercase?

**Review by ferreiratiago — CHANGES_REQUESTED** [review]

File: plugins/aladino/functions/toNumber_test.go (hunk 1 above)
Comment (ferreiratiago) [reply plugins/aladino/functions/toNumber_test.go:29]: I see. These are the error properties.

**Review by ferreiratiago — APPROVED** [review]

File: plugins/aladino/functions/toNumber_test.go (hunk 2 above)
Comment (zolamk) [reply plugins/aladino/functions/toNumber_test.go:31]: those are the errors properties

**Pull request merged**